core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -463,4 +475,171 @@ public void testRecursiveTimeFilterRegex() throws Exception {
     Assert.assertTrue(outputValue.contains("Hello,World"));
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
+
+  @Test
+  public void testFileBatchInputFormatText() throws Exception {
+    String filePath = "file:///tmp/test/test.txt";
+    String testData = "String for testing purposes";
+
+    Path textFile = new Path(filePath);
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+    FSDataOutputStream writeData = fs.create(textFile);
+    writeData.write(testData.getBytes());
+    writeData.flush();
+    writeData.close();
+
+    ETLStage source = new ETLStage(
+      "source", new ETLPlugin("File", BatchSource.PLUGIN_TYPE,
+                              ImmutableMap.<String, String>builder()
+                                .put(Constants.Reference.REFERENCE_NAME, "TestFile")
+                                .put(Properties.File.FILESYSTEM, "Text")
+                                .put(Properties.File.PATH, filePath)
+                                .put(Properties.File.FILE_SOURCE_FORMAT, "text")
+                                .put(Properties.File.IGNORE_NON_EXISTING_FOLDERS, "false")
+                                .build(),
+                              null));
+
+    String outputDatasetName = "test-filesource-text";
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(outputDatasetName));
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(BATCH_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("FileSourceTextFormat");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForRuns(ProgramRunStatus.COMPLETED, 1, 2, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
+    List<StructuredRecord> output = MockSink.readOutput(outputManager);
+
+    Assert.assertEquals(1, output.size());
+    Assert.assertEquals(testData, output.get(0).get("body"));
+  }
+
+  @Test
+  public void testFileBatchInputFormatAvro() throws Exception {
+    Schema recordSchema = Schema.recordOf("record",
+                                          Schema.Field.of("i", Schema.of(Schema.Type.INT)),
+                                          Schema.Field.of("l", Schema.of(Schema.Type.LONG)));
+
+    String filePath = "file:///tmp/test/test.avro";
+    Path avroFilePath = new Path(filePath);
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+
+    ETLStage source = new ETLStage(
```

Comment:
half of these test cases are exactly the same and can be refactored into a common method. Each test case should only deploy the app, write input, then call that common method.